lib/ansible/module_utils/cloud.py

```diff
@@ -106,3 +159,62 @@ def retry_func(*args, **kwargs):
             return retry_func  # true decorator
 
         return deco
+
+    @classmethod
+    def exponential_backoff(cls, retries=10, delay=3, backoff=2, max_delay=60):
+        """
+        Retry calling the Cloud decorated function using an exponential backoff.
+
+        Kwargs:
+            retries (int): Number of times to retry a failed request before giving up
+                default=10
+            delay (int or float): Initial delay between retries in seconds
+                default=3
+            backoff (int or float): backoff multiplier e.g. value of 2 will
+                double the delay each retry
+                default=1.1
+            max_delay (int or None): maximum amount of time to wait between retries.
+                default=60
+        """
+        return cls._backoff(_exponential_backoff(
+            retries=retries, delay=delay, backoff=backoff, max_delay=max_delay))
+
+    @classmethod
+    def jittered_backoff(cls, retries=10, delay=3, max_delay=60):
+        """
+        Retry calling the Cloud decorated function using a jittered backoff
+        strategy. More on this strategy here:
+
+        https://www.awsarchitectureblog.com/2015/03/backoff.html
+
+        Kwargs:
+            retries (int): Number of times to retry a failed request before giving up
+                default=10
+            delay (int): Initial delay between retries in seconds
+                default=3
+            max_delay (int): maximum amount of time to wait between retries.
+                default=60
+        """
+        return cls._backoff(_full_jitter_backoff(
+            retries=retries, delay=delay, max_delay=max_delay))
+
+    @classmethod
+    def backoff(cls, tries=10, delay=3, backoff=1.1):
```

Comment:
I left `backoff` untouched because I didn't want to break any 3rd party modules that currently use it.

While the `backoff` method is an exponential backoff its defaults don't provide the behavior that you would expect with such an algorithm. The duplication is minimal in that `backoff` simply calls out to `exponential_backoff`. The `backoff` method using defaults more or less boils down to the following delays:

```
>>> ['%02.2f' % (3 * 1.1 ** i) for i in range(0, 10)]
['3.00', '3.30', '3.63', '3.99', '4.39', '4.83', '5.31', '5.85', '6.43', '7.07']
```

This isn't the typical exponential backoff pattern that you'd expect.

`exponential_backoff` aims to provide more aggressive backoffs by default with the option to cap the maximum time each retry waits. e.g.

```
>>> list(_exponential_backoff()())
[2, 4, 8, 16, 32, 60, 60, 60, 60, 60]
```